metals/src/main/scala/scala/meta/internal/metals/testProvider/TestSuitesFinder.scala

```diff
@@ -0,0 +1,115 @@
+package scala.meta.internal.metals.testProvider
+
+import scala.meta.internal.implementation.ImplementationProvider
+import scala.meta.internal.metals.BuildTargets
+import scala.meta.internal.metals.DefinitionProvider
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.debug.BuildTargetClasses
+
+import ch.epfl.scala.bsp4j.BuildTarget
+
+trait TestSuitesFinder {
+  def findTestSuites(): Seq[TestSuiteDiscoveryResult]
+}
+
+final class TestSuitesFinderImpl(
+    buildTargets: BuildTargets,
+    buildTargetClasses: BuildTargetClasses,
+    definitionProvider: DefinitionProvider,
+    implementationProvider: ImplementationProvider
+) extends TestSuitesFinder {
+
+  override def findTestSuites(): Seq[TestSuiteDiscoveryResult] = {
+    buildTargets.allBuildTargetIds.toList
+      .flatMap(buildTargets.info)
+      .filterNot(_.getDisplayName.endsWith("build"))
+      .map(buildTarget => createTestDiscoveryFor(buildTarget))
+      // don't send empty results
+      .filter(_.discovered.asScala.exists(_.nonEmpty))
+  }
+
+  private def createTestDiscoveryFor(buildTarget: BuildTarget) = {
+    val classes = buildTargetClasses
+      .classesOf(buildTarget.getId)
+      .testClasses
+      // filter out symbols which don't represent classes
```

Comment:
I think there can be tests which are objects, no ? Is there a need for this filtering?

Reply:
I'll have to look at it once again, but from what I remember every suite was listed twice. Once as a class and te second time as an object :/

Reply:
Ok, so here is how it looks like for most popular frameworks like scalatest, junit, munit and utest. There are two symbols, one for class and one for an object which corresponds to the same class name. However, only one of them has a location in code -- and that one will be kept while the second one will be filtered out.
```shell
"a/MunitTestSuite., a.MunitTestSuite"
"a/MunitTestSuite#, a.MunitTestSuite"
```